### PR TITLE
Update pagination documentation

### PR DESCRIPTION
**Pull request by JSFernandes**
This PR updates the documentation with the new pagination strategy which we hope to release soon.

I was considering updating the changelog for the documentation, but it includes a lot of irrelevant Slate changes. Shall we reboot it?

### CODE REVIEW

File: source/includes/_setup.md [hunk 1]
```diff
@@ -90,25 +90,40 @@ If you exceed the limit you will receive a `HTTP 429 (Too Many Requests)`.
...
+  -u YOUR_API_KEY:x
+```
+
+As of API version `20210316`, pagination is performed with a `starting_after` parameter. This parameter takes an existing object ID value and returns objects listed after the named object, in reverse chronological order.
+
+The HTTP header `X-Pages-HasMore` indicates whether more records can be fetched by using the same query with a lower `starting_after`.
 
 Bear in mind that Quaderno paginates `GET` index results.
 
 You can change the number of objects to be returned with the `limit` parameter, defaulting to `25`.
```
Comment (polimorfico) [inline source/includes/_setup.md:113]: Should be set a maximum number of objects per page to avoid abuses?

File: source/includes/_setup.md [hunk 2]
```diff
@@ -90,25 +90,40 @@ If you exceed the limit you will receive a `HTTP 429 (Too Many Requests)`.
 
 # Pagination
 
-> A call with the `page` parameter set:
+> A call with the `max_id` parameter set:
```
Comment (polimorfico) [inline source/includes/_setup.md:93]: It should be `starting_after`

File: source/includes/_setup.md [hunk 3]
```diff
@@ -90,25 +90,40 @@ If you exceed the limit you will receive a `HTTP 429 (Too Many Requests)`.
...
 
 ```shell
-curl https://ACCOUNT_NAME.quadernoapp.com/api/contacts.json?page=2 \ 
+curl https://ACCOUNT_NAME.quadernoapp.com/api/contacts.json?max_id=42 \
```
Comment (polimorfico) [inline source/includes/_setup.md:96]: The name of the query string should be `starting_after`

**Review by polimorfico — CHANGES_REQUESTED** [review]

File: source/includes/_setup.md (hunk 3 above)
Comment (jrub) [reply source/includes/_setup.md:96]: There's another reference to the `page` parameter on Line 170 we should change

File: source/includes/_setup.md [hunk 4]
```diff
@@ -90,25 +90,40 @@ If you exceed the limit you will receive a `HTTP 429 (Too Many Requests)`.
...
 ```
 
-These HTTP headers inform you about the page context:
+> DEPRECATED: A call with the `page` parameter set:
 
-- `X-Pages-CurrentPage`
-- `X-Pages-TotalPages`
+```shell
+curl https://ACCOUNT_NAME.quadernoapp.com/api/contacts.json?page=2 \
+  -u YOUR_API_KEY:x
```
Comment (jrub) [inline source/includes/_setup.md:104]: I would move this example to get it visually to the right of the legacy note, or at least change DEPRECATED to **Legacy**. Note there's a code suggestion below.

**Review by jrub — COMMENTED** [review]
Added some code suggestions for unifying to "legacy" (already used in the Tax API) instead of deprecated. It could be the other way around if preferred.

Also, reviewed the changelog searching for these Slate changes you mentioned, but couldn't find any.

File: source/includes/_setup.md (hunk 1 above)
Comment (JSFernandes) [reply source/includes/_setup.md:113]: There is, but it looks like we didn't document it. Going to add it.

**Comment by JSFernandes** [comment]
@jrub The Slate changes I meant are the ones in https://github.com/quaderno/quaderno-api/blob/master/CHANGELOG.md
Version 2.7.0 was added by us and contains relevant documentation changes. Every previous version has changes related with slate infrastructure/environment/setup.

**Comment by polimorfico** [comment]
@JSFernandes Feel free to remove all the content in changelog that are not related to our API.

**Comment by jrub** [comment]
Oh, I see what you meant with the changelog now. We definitely need a changelog for our public API (currently `source/includes/_changelog.md`, as is used for the online dev docs), but, do we really need another for tech changes over the repo? (which I would say it's usually the use of the one on the root of an open source project)

If we do then it should be about our tech stack changes and similar, not Slate's, that is for sure.

**Comment by JSFernandes** [comment]
@jrub @polimorfico Can you please take a look at the recent changes? I hope to deploy the pagination changes and this today. I will then work on updating the gem.

File: source/includes/_changelog.md [hunk 5]
```diff
@@ -2,7 +2,11 @@
 
 ## API versions and changes
 
-### 20170914 (Current version)
+### 20210316 (Current version)
+* Breaking change: record pagination is no longer performed with a `page` parameter, but with a `max_id` parameter. See the [pagination documentation](https://developers.quaderno.io/api/#pagination) for more details.
```
Comment (polimorfico) [inline source/includes/_changelog.md:6]: The name of the new parameter is wrong.

**Review by polimorfico — APPROVED** [review]
I only found a small typo. The rest is okay IMO.

**Comment by JSFernandes** [comment]
@polimorfico can you please merge and deploy when you have time?

**Pull request merged**